### PR TITLE
Migrate log screen to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/logging/logging_screen.dart [hunk 1]
```diff
@@ -47,7 +45,7 @@ class LoggingScreen extends Screen {
 
     return StreamBuilder<String>(
       initialData: controller.statusText,
-      stream: controller.onLogStatusChanged,
+      stream: controller.onLogStatusChanged as Stream<String>?,
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/logging_screen.dart:48]: if you give onLogStatusChanged type `Stream<String>` I think you can remove this cast

File: packages/devtools_app/lib/src/screens/logging/logging_screen.dart [hunk 2]
```diff
@@ -77,11 +75,11 @@ Example queries:
 
 class _LoggingScreenState extends State<LoggingScreenBody>
     with AutoDisposeMixin, SearchFieldMixin<LoggingScreenBody> {
-  LogData selected;
+  LogData? selected;
 
-  LoggingController controller;
+  LoggingController? controller;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/logging_screen.dart:80]: use initialized pattern and make this late

File: packages/devtools_app/lib/src/screens/logging/logging_screen.dart [hunk 3]
```diff
@@ -176,10 +174,10 @@ class _LoggingScreenState extends State<LoggingScreenBody>
   void _showFilterDialog() {
     showDialog(
       context: context,
-      builder: (context) => FilterDialog<LoggingController, LogData>(
+      builder: (context) => FilterDialog<LoggingController?, LogData>(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/logging_screen.dart:177]: non nullable

File: packages/devtools_app/lib/src/screens/logging/logging_screen.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_screen.dart:48]: done

File: packages/devtools_app/lib/src/screens/logging/logging_screen.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_screen.dart:80]: done

File: packages/devtools_app/lib/src/screens/logging/logging_screen.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_screen.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**